### DISCUSSION TITLE
[RFC] ConfigProvider support scope

### DISCUSSION THREAD

**Discussion by crazyair** [ant-design/ant-design#50584] [RFCs]
# RFC

## Summary

目前 `ConfigProvider` 不支持作用域控制，比如这里需要用默认的 20px，但是实际只能是 12px
![image](https://github.com/user-attachments/assets/7edc6f93-08cf-44ae-b870-92681852787a)
https://stackblitz.com/edit/react-qb3j44?file=demo.tsx

## Motivation

`ConfigProvider` 支持作用域，可以控制 theme 的配置只生效当前子组件，当遇到子组件有 `ConfigProvider` 则失效

## Detailed design

```tsx
<ConfigProvider scope={0} disabled thene={{ token:{ fontSize: 20 } }}>
    // 此处组建禁用、文字为 20px
    <ConfigProvider>
         // 此处组建不禁用、文字是默认的 14px
    </ConfigProvider>
</ConfigProvider>
```

**Comment by MadCcc** [comment]
还是觉得不太符合正常使用的习惯，参考一下单独配置组件 token 吧
